### DISCUSSION TITLE
Remove `:` (colon) from all internal cache keys to comply with PSR-16/PSR-6

### DISCUSSION THREAD

**Discussion by eldair** [laravel/framework#51663] [Ideas]
I think `:` (colon) should be removed and perhaps replaced with `.` or `-` from all internal cache keys (eg. `illuminate:foundation:down`) so that it becomes compliant with PSR-16/PSR-6.
I came upon this issue when writing redis cache adapter based on symfony/cache (for better tags support) where exception is thrown when the key does not comply with PSR rules and all internal cache keys follow pattern with `:`

**Comment by shaedrich** [comment]
I imagine, this could be a breaking chance, so the team might be reluctant to take this on 🤔

**Reply by henzeb** [reply]
But I'd you really want to, you can do that in the adapter itself...

**Reply by eldair** [reply]
I did make "clean key" in my adapter to replace those characters for every store and fetch from cache but it should be resolved in the framework itself.
I am not really sure why would it be a breaking change, those strings are not really used in userland by themselves or maybe I'm overseeing something.

**Reply by shaedrich** [reply]
Well, Taylor is hestitant to take _any_ changes by people he doesn't particularily like. If there's something, he can reject for pointing towards the "wrong" version, he'll be quick to do so. I just wanted to warn you that this might happen 😉

**Reply by henzeb** [reply]
> I did make "clean key" in my adapter to replace those characters for every store and fetch from cache but it should be resolved in the framework itself. I am not really sure why would it be a breaking change, those strings are not really used in userland by themselves or maybe I'm overseeing something.

the breaking change is that those keys are being used under the hood by the public.  Imagine you have your car stored on a huge parking lot at the character F, and suddenly someone changes it to pictures of animals, non alphabetically. You'd need a map that says that F is a Grizzly bear.  

If after an update the key would become `illuminate.foundation.down` the application won't be finding `illuminate:foundation:down` and thus respond that this key doesn't exist, while in reality it does. 

You can try for 11.x.

**Reply by eldair** [reply]
> > I did make "clean key" in my adapter to replace those characters for every store and fetch from cache but it should be resolved in the framework itself. I am not really sure why would it be a breaking change, those strings are not really used in userland by themselves or maybe I'm overseeing something.
> 
> the breaking change is that those keys are being used under the hood by the public. Imagine you have your car stored on a huge parking lot at the character F, and suddenly someone changes it to pictures of animals, non alphabetically. You'd need a map that says that F is a Grizzly bear.
> 
> If after an update the key would become `illuminate.foundation.down` the application won't be finding `illuminate:foundation:down` and thus respond that this key doesn't exist, while in reality it does.
> 
> You can try for 11.x.

Yeah, you are right. Maybe it would make more sense if they would be willing to completely switch to symfony/cache, I'll think about it.

**Reply by eldair** [reply]
> Well, Taylor is hestitant to take _any_ changes by people he doesn't particularily like. If there's something, he can reject for pointing towards the "wrong" version, he'll be quick to do so. I just wanted to warn you that this might happen 😉

Yeah, you are right about that one
